src/test/scala/is/hail/distributedmatrix/BlockMatrixSuite.scala

```diff
@@ -686,52 +701,69 @@ class BlockMatrixSuite extends SparkSuite {
       } {
         val filteredViaBlock = bm.filter(keep.map(_.toLong), keep.map(_.toLong)).toLocalMatrix()
         val filteredViaBreeze = lm(keep.toIndexedSeq, keep.toIndexedSeq).copy
-        
+
         assert(filteredViaBlock === filteredViaBreeze)
       }
     }
   }
-  
+
   @Test
   def filter() {
     val lm = new BDM[Double](9, 10, (0 until 90).map(_.toDouble).toArray)
 
-    for { blockSize <- Seq(1, 2, 3, 5, 10, 11)
+    for {blockSize <- Seq(1, 2, 3, 5, 10, 11)
     } {
-      val bm = BlockMatrix.from(sc, lm, blockSize)      
+      val bm = BlockMatrix.from(sc, lm, blockSize)
       for {
         keepRows <- Seq(
           Array(1),
           Array(0, 3, 4, 5, 7),
-          Array(0, 1, 2, 3, 4, 5, 6, 7, 8))  
+          Array(0, 1, 2, 3, 4, 5, 6, 7, 8))
         keepCols <- Seq(
           Array(2),
           Array(1, 4, 5, 7, 8, 9),
-          Array(0, 1, 2, 3, 4, 5, 6, 7, 8, 9))  
+          Array(0, 1, 2, 3, 4, 5, 6, 7, 8, 9))
       } {
         val filteredViaBlock = bm.filter(keepRows.map(_.toLong), keepCols.map(_.toLong)).toLocalMatrix()
         val filteredViaBreeze = lm(keepRows.toIndexedSeq, keepCols.toIndexedSeq).copy
-        
+
         assert(filteredViaBlock === filteredViaBreeze)
       }
     }
   }
-  
+
   @Test
   def random() {
     var lm1 = BlockMatrix.random(hc, 5, 10, 2, seed = 1).toLocalMatrix()
     var lm2 = BlockMatrix.random(hc, 5, 10, 2, seed = 1).toLocalMatrix()
     var lm3 = BlockMatrix.random(hc, 5, 10, 2, seed = 2).toLocalMatrix()
-    
+
     assert(lm1 === lm2)
     assert(lm1 !== lm3)
     assert(lm1.data.forall(x => x >= 0 && x <= 1))
-    
+
     lm1 = BlockMatrix.random(hc, 5, 10, 2, seed = 1, gaussian = true).toLocalMatrix()
     lm2 = BlockMatrix.random(hc, 5, 10, 2, seed = 1, gaussian = true).toLocalMatrix()
     lm3 = BlockMatrix.random(hc, 5, 10, 2, seed = 2, gaussian = true).toLocalMatrix()
-    
+
     assert(lm1 === lm2)
     assert(lm1 !== lm3)
   }
+
+  @Test
+  def testMelt(): Unit = {
```

Comment:
Make a larger matrix and test this for a few different block sizes using a `for`:
```
    val lm = new BDM[Double](9, 10, (0 until 90).map(_.toDouble).toArray)
    val expected = ...
    
    for { blockSize <- Seq(1, 4, 10) } { ... }
```